src/states/bootstrapping.rs

```diff
@@ -225,19 +225,20 @@ impl Bootstrapping {
     }
 
     fn handle_timeout(&mut self, token: u64) {
+        let mut retry_bootstrap = false;
         if let Some((ref bootstrap_id, bootstrap_token)) = self.bootstrap_connection {
             if bootstrap_token == token {
                 debug!(
                     "{:?} Timeout when trying to bootstrap against {:?}.",
                     self, bootstrap_id
                 );
-            } else {
-                return;
+                retry_bootstrap = true;
```

Comment:
why not just `return self.rebootstrap();` right here ?

Reply:
Because borrow checker won't allow us to use `self` in this place since it's borrowed in this context :/